### PR TITLE
Adjust dependency confusion doc for Cachi2

### PR DESCRIPTION
**Pull request by chmeliik**
CLOUDBLD-11125

More details in commit messages

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- n/a Code coverage from testing does not decrease and new code is covered
- n/a New code has type annotations
- [x] Docs updated (if applicable)

### CODE REVIEW

File: docs/dependency_confusion.md [hunk 1]
```diff
@@ -19,127 +18,128 @@ of the internal one.
...
-confusion during a *rebuild*. If you build the same revision of your repository, or a different
-revision which uses the same versions of dependencies, Cachito will use the dependencies that it has
-already downloaded. Dependency confusion should only be a concern when building for the first time
-or when changing the version of a dependency.
-
 # Package Managers
 
-*Sometimes users may state that Cachito supports a certain programming language. This is not
-entirely accurate. Cachito only supports package managers. The ecosystem of a programming language
+*Sometimes users may state that Cachi2 supports a certain programming language. This is not
```
Comment (brunoapimentel) [inline docs/dependency_confusion.md:47]: It doesn't hurt to leave this here, but I wonder if this disclaimer is needed at all in this doc.

**Comment by brunoapimentel** [comment]
On a side note, we should use this PR as an example of how to properly break up your commits :smile:

File: docs/dependency_confusion.md (hunk 1 above)
Comment (chmeliik) [reply docs/dependency_confusion.md:47]: True, I'll remove this

File: docs/dependency_confusion.md [hunk 2]
```diff
@@ -19,127 +18,123 @@ of the internal one.
...
-define the name of your module as well as the names and versions of its direct dependencies.
-Thanks to golang’s unique system of dependency management, this file is enough to ensure
-reproducible builds on its own.
+define the name of your module as well as the names and versions of its dependencies. Thanks to
+golang’s unique system of dependency management, this file is enough to ensure reproducible builds
+on its own.
 
 However, due to golang versions being tied to git tags, you are technically at the mercy of
 repository maintainers. To prevent surprises, whenever go downloads any dependencies, it
 saves their checksums in a [go.sum](https://golang.org/ref/mod#go) file. This file is later used to
```
Comment (taylormadore) [inline docs/dependency_confusion.md:58]:
```suggestion
saves their checksums in a [go.sum](https://go.dev/ref/mod#go-sum-files) file. This file is later used to
```

**Comment by taylormadore** [comment]
Other than the small go.sum link update I noted above, looks great!

File: docs/dependency_confusion.md (hunk 2 above)
Comment (chmeliik) [reply docs/dependency_confusion.md]: Nice catch, updated :+1:

**Review by taylormadore — APPROVED** [review]

**Pull request merged**